### ISSUE TITLE
UCI Net Images for Emotion Analysis

### ISSUE THREAD

**Issue by muni2773** [davidsandberg/facenet#698]
Hi, I just ran VGG Face 2 trainer/classifier on the images labeled for 4 emotions, Angry, Happy, Sad and Neutral. Using the split data set I am getting an accuracy of only 25%. Any thought on how I can improve that ?

Also any chance of releasing a inception resnet-v2 frozen model ? I have 2 Pascal GPUs if you need any HW horsepower. 

Regards.

**Comment by speculaas** [comment]
Dear Muni, 

I just tried a similar classification like what you did.

I have been studying emotion recognition recently. 

I also used vggface-2 pretrained model to do feature extraction for 8 emotions as in CK+ dataset. 

After classifier training, I ran testing : "python src/classifier.py CLASSIFY" and got "Accuracy: 0.267".

Do you or anybody has any idea why "model = SVC(kernel='linear')" cannot do well in this case?

http://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html

Is it because linear classifier not good?
Or is it because feature extraction is not good?

I am also trying to fine tune the vgg2-inception-resnet pretrained model using a combination of CK+, JAFFE, TFEID datasets as training set. 
Then maybe use this fine-tuned inception-resnet to do feature extraction and see what happens. 

Also, up till now, I studied the following materials: 

https://arxiv.org/abs/1804.08348
This paper gives a survey.

https://arxiv.org/abs/1609.06591
This paper use conv layers of face recognizing net plus a new fully connected layer to do emotion recognition

https://kpzhang93.github.io/papers/icmi.pdf
This paper describing their EmotiW 2017 winning method says before fine tuning for group emotion recognition, they first trained Resnet64 on Webface then finetune on ExpW.

BR,
Jimmy
